### PR TITLE
Import preact devtools renderer code

### PR DESCRIPTION
**Pull request by marvinhagemeister**
This is an integration for our [custom devtools](https://github.com/preactjs/preact-devtools) :tada: 

Screenshot from 2019-11-24 22-34-26

**Tasks:**

- [x] Convert tests written in TS to JS
- [x] Fix infinite loop in `jsonify` with some `Suspense` components in demo app
- [x] Fix `createContext` value always parsed as an `object` 
- [x] Code Coverage :grimacing:

### CODE REVIEW

**Comment by marvinhagemeister** [comment]
I'm unsure if I should move it to `preact/devtools` with this PR or with a separate one. Leaving it in `preact/debug` for now would have the advantage of keeping the existing imports the same for users.

**Comment by marvinhagemeister** [comment]
I think this PR is ready to review since a few days. If there is anything to make reviewing it easier, please reach out.

File: debug/src/devtools/10/IdMapper.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	/** @type {Map<any, number>} */
+	const instToId = new Map();
+
+	/** @type {Map<number, import('../../internal').VNode>} */
+	const idToVNode = new Map();
+
+	/** @type {Map<number, any>} */
+	const idToInst = new Map();
+
+	let uuid = 1;
```
Comment (JoviDeCroock) [inline debug/src/devtools/10/IdMapper.js]: NIT: can we call this id instead, `uuid` feels like the long string

File: debug/src/devtools/10/options.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	let prevBeforeUnmount = options.unmount;
+	let prevBeforeDiff = options._diff;
+	let prevAfterDiff = options.diffed;
+
+	options.vnode = vnode => {
+		// Tiny performance improvement by initializing fields as doubles
+		// from the start. `performance.now()` will always return a double.
+		// See https://github.com/facebook/react/issues/14365
+		// and https://slidr.io/bmeurer/javascript-engine-fundamentals-the-good-the-bad-and-the-ugly
+		vnode.startTime = NaN;
```
Comment (JoviDeCroock) [inline debug/src/devtools/10/options.js]: Today I learned, thanks!

File: debug/src/devtools/10/vnode.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import { Fragment } from 'preact';
+
+/**
+ * Check if a `vnode` is the root of a tree
+ * @param {import('../../internal').VNode} vnode
+ * @returns {boolean}
+ */
+export function isRoot(vnode) {
+	return vnode._parent == null && vnode.type === Fragment;
```
Comment (JoviDeCroock) [inline debug/src/devtools/10/vnode.js]: Makes me wonder if we need a special indicator for this node

Comment (marvinhagemeister) [reply debug/src/devtools/10/vnode.js]: Yeah, true, we may need something to mark root nodes as roots instead of plain `Fragments` in the future.

**Pull request merged**